noir_stdlib/src/hash/mimc.nr

```diff
@@ -0,0 +1,666 @@
+        6129250029437675212264306655559561251995722990149771051304736001195288083309,
+        10773245783118750721454994239248013870822765715268323522295722350908043393604,
+        4490242021765793917495398271905043433053432245571325177153467194570741607167,
+        19596995117319480189066041930051006586888908165330319666010398892494684778526,
+        837850695495734270707668553360118467905109360511302468085569220634750561083,
+        11803922811376367215191737026157445294481406304781326649717082177394185903907,
+        10201298324909697255105265958780781450978049256931478989759448189112393506592,
+        13564695482314888817576351063608519127702411536552857463682060761575100923924,
+        9262808208636973454201420823766139682381973240743541030659775288508921362724,
+        173271062536305557219323722062711383294158572562695717740068656098441040230,
+        18120430890549410286417591505529104700901943324772175772035648111937818237369,
+        20484495168135072493552514219686101965206843697794133766912991150184337935627,
+        19155651295705203459475805213866664350848604323501251939850063308319753686505,
+        11971299749478202793661982361798418342615500543489781306376058267926437157297,
+        18285310723116790056148596536349375622245669010373674803854111592441823052978,
+        7069216248902547653615508023941692395371990416048967468982099270925308100727,
+        6465151453746412132599596984628739550147379072443683076388208843341824127379,
+        16143532858389170960690347742477978826830511669766530042104134302796355145785,
+        19362583304414853660976404410208489566967618125972377176980367224623492419647,
+        1702213613534733786921602839210290505213503664731919006932367875629005980493,
+        10781825404476535814285389902565833897646945212027592373510689209734812292327,
+        4212716923652881254737947578600828255798948993302968210248673545442808456151,
+        7594017890037021425366623750593200398174488805473151513558919864633711506220,
+        18979889247746272055963929241596362599320706910852082477600815822482192194401,
+        13602139229813231349386885113156901793661719180900395818909719758150455500533,
+    ];
+
+    let mut r = 0;
+    for elem in array {
+        let h = mimc(elem, r, constants, exponent);
+        r = r + elem + h;
+    }
+    r
+}
+
+// MiMC-2p/p implementation
+// constants are (publicly generated) random numbers, for instance using keccak as a ROM.
+// You must use constants generated for the native field
+// Rounds number should be ~ log(p)/log(exp) log_5(21888242871839275222246405745257275088548364400416034343698204186575808495617) ~= 110
+// For 254 bit primes, exponent 5 and 220 rounds seems to be recommended
+
+
+
+fn MiMCFeistel(xL_in : Field, xR_in : Field, k : Field) ->  [Field;2] {
+
+    let NUM_ROUNDS = 220;
+
+    //xL <-- xLxR[0]
+    //xR <-- xLxR[1]
+    let mut xLxR: [Field;2] = [0;2];  
+    let constant: [Field; 218] = [
```

Comment:
Is this the same constants list as Feistel decryption? If so could we move it to a global?

Reply:
Ideally at the bottom of the file so that we can see the algorithm implementations first